internal/services/resource/resource_deployment_script_common.go

```diff
@@ -88,6 +83,13 @@ func getDeploymentScriptArguments(kind DeploymentScriptKind) map[string]*plugins
 			ValidateFunc: validate.ISO8601DurationBetween("PT1H", "P1DT2H"),
 		},
 
+		"version": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
```

Comment:
Rather than dropping the validation, we should make sure the version string is in the expected format, e.g. `x.y` for PowerShell or `x.y.z` for Azure CLI.